### PR TITLE
Renamed browser-side to main process

### LINKED ISSUE #1287: RFC: The names 'client-side' and 'browser-side' are confusing
Requesting comments.

When I was reading through atom-shell's documentation, and I found the terms `client-side` (sometimes `renderer`) and `browser-side` confusing. (E.g. In [docs/tutorial/quick-start.md](https://github.com/atom/atom-shell/blob/master/docs/tutorial/quick-start.md#the-browser-side).)

Is this just me?

I suggest the terms `front-end` and `back-end`. This sticks with the client-server analogy, and is less confusing (IMO).

I volunteer to make a PR to the docs if people think this is reasonable.

### PR DESCRIPTION
**Pull request by ArtskydJ**
- Changed occurrences of "browser process" and "browser-side" to "main process".
- Changed a few occurrences of "web page" to "renderer". There are still many occurrences of web page. (Some occurrences were more fitting than others.)
- Changed a few files that had "browser" in their name to "main-process"
- Fixed various typos that I found.

Fixes #1287

### CODE REVIEW

**Comment by bwin** [comment]
Are we really done here already? (Just asking)

**Comment by bwin** [comment]
Sry, I'm in mobile, wanted to comment in the issue...

**Comment by mark-hahn** [comment]
It works for me.

File: docs/api/browser-window.md [hunk 1]
```diff
@@ -241,16 +241,16 @@ Get the unique ID of this window.
 ### BrowserWindow.destroy()
 
 Force closing the window, the `unload` and `beforeunload` event won't be emitted
-for the web page, and `close` event would also not be emitted for this window,
-but it would guarantee the `closed` event to be emitted.
+for the renderer process (web page), and `close` event would also not be emitted
```
Comment (zcbenz) [inline docs/api/browser-window.md]: We should only use `web page` here, since it is about DOM events.

File: docs/api/browser-window.md [hunk 2]
```diff
@@ -241,16 +241,16 @@ Get the unique ID of this window.
...
 
-You should only use this method when the web page has crashed.
+You should only use this method when the renderer process (web page) has crashed.
 
 ### BrowserWindow.close()
 
 Try to close the window, this has the same effect with user manually clicking
-the close button of the window. The web page may cancel the close though, see
-the [close event](window#event-close).
+the close button of the window. The renderer process (web page) may cancel the
```
Comment (zcbenz) [inline docs/api/browser-window.md]: Only `web page` here.

File: docs/api/browser-window.md [hunk 3]
```diff
@@ -327,11 +327,11 @@ Returns an array that contains window's width and height.
 * `width` Integer
 * `height` Integer
 
-Resizes the window's client area (e.g. the web page) to `width` and `height`.
+Resizes the renderer's area (i.e. the web page) to `width` and `height`.
```
Comment (zcbenz) [inline docs/api/browser-window.md]: The `client area` is a desktop development term, and should not be replaced with `renderer`.

File: docs/api/browser-window.md [hunk 4]
```diff
@@ -327,11 +327,11 @@ Returns an array that contains window's width and height.
...
 
 ### BrowserWindow.getContentSize()
 
-Returns an array that contains window's client area's width and height.
+Returns an array that contains the renderer's width and height.
```
Comment (zcbenz) [inline docs/api/browser-window.md]: No `renderer` here.

File: docs/api/ipc-main-process.md [hunk 5]
```diff
@@ -45,4 +46,4 @@ Assign to this to return an value to synchronous messages.
 
 ### Event.sender
 
-The `WebContents` of the web page that has sent the message.
+The `WebContents` of the renderer that has sent the message.
```
Comment (zcbenz) [inline docs/api/ipc-main-process.md]: It is possible for a renderer to have multiple `WebContents`, we should just remove "of the renderer".

File: docs/tutorial/quick-start.md [hunk 6]
```diff
@@ -11,26 +11,18 @@ It doesn't mean atom-shell is a JavaScript binding to GUI libraries. Instead,
...
+### The main process
 
-If you have experience with Node.js web applications, you will know that there
-are two types of JavaScript scripts: the server side scripts and the client side
-scripts. Server-side JavaScript is that which runs on the Node.js
-runtime, while client-side JavaScript runs inside the user's browser.
+The main atom-shell process displays a GUI by showing web pages. We have
+**scripts that run in the atom-shell runtime**, that create **scripts
+that run in the web page**. We call them **main process scripts**,
+and **renderer process scripts**.
```
Comment (zcbenz) [inline docs/tutorial/quick-start.md]: I think we can say main process creates web pages, and web pages run in their own renderer processes. We can ignore the "renderer process scripts" part to make things simpler.

**Comment by ArtskydJ** [comment]
@zcbenz I reverted the ones you mentioned, and updated the introduction paragraph.

Are lines 228 and 247 in quick-start.md correct?

**Comment by zcbenz** [comment]
It looks good to me, thanks!

**Pull request merged**